server/app/services/barman-service.js

```diff
@@ -144,6 +151,12 @@ async function updateBarmanById(barmanId, updatedBarman, _embedded) {
 
   if (!currentBarman) throw createUserError('UnknownBarman', 'This Barman does not exist');
 
+  if ((updatedBarman.leaveAt
+    && new Date(currentBarman.createdAt) > new Date(currentBarman.leaveAt))
+    || Date.now() < new Date(updatedBarman.leaveAt)) {
```

Comment:
Same goes for leaveAt here. It should already be a date (thanks joi)

Reply:
And this check will not work as Date.now() is a timestamp and new Date() is a Date

Reply:
Also this can be done before through joi: https://github.com/hapijs/joi/blob/v13.7.0/API.md#datemaxdate